luigi/contrib/redshift.py

```diff
@@ -37,6 +42,85 @@
                    "Will crash at runtime if postgres functionality is used.")
 
 
+class _CredentialsMixin():
+    """
+    This mixin is used to provide the same credential properties
+    for AWS to all Redshift tasks. It also provides a helper method
+    to generate the credentials string for the task.
+    """
+
+    @property
+    def aws_access_key_id(self):
+        """
+        Override to return the key id.
+        """
+        return self._get_s3_configuration_attribute('aws_access_key_id')
+
+    @property
+    def aws_secret_access_key(self):
+        """
+        Override to return the secret access key.
+        """
+        return self._get_s3_configuration_attribute('aws_secret_access_key')
+
+    @property
+    def aws_account_id(self):
+        """
+        Override to return the account id.
+        """
+        return None
```

Comment:
I was really trying to mimic the existing functionality for those fields, but would be happy to add that in if you think it is the right thing to do.

Right now, I'm not to sure that the implementation (current and previous) was even really so correct, because it is using the information from the `s3` section as opposed to a `redshift` section in the configuration file.